os/arch/arm/src/imxrt/imxrt_ehci.c

```diff
@@ -0,0 +1,5148 @@
+#else
+	return (uint16_t) addr[1] << 8 | (uint16_t) addr[0];
+#endif
+}
+
+/****************************************************************************
+ * Name: imxrt_read32
+ *
+ * Description:
+ *   Read 32-bit little endian data
+ *
+ ****************************************************************************/
+static inline uint32_t imxrt_read32(const uint8_t *addr)
+{
+#ifdef CONFIG_ENDIAN_BIG
+	return (uint32_t) imxrt_read16(&addr[0]) << 16 | (uint32_t) imxrt_read16(&addr[2]);
+#else
+	return (uint32_t) imxrt_read16(&addr[2]) << 16 | (uint32_t) imxrt_read16(&addr[0]);
+#endif
+}
+
+/****************************************************************************
+ * Name: imxrt_write16
+ *
+ * Description:
+ *   Write 16-bit little endian data
+ *
+ ****************************************************************************/
+
+#if 0							/* Not used */
+static void imxrt_write16(uint16_t memval, uint8_t *addr)
+{
+#ifdef CONFIG_ENDIAN_BIG
+	addr[0] = memval & 0xff;
+	addr[1] = memval >> 8;
+#else
+	addr[0] = memval >> 8;
+	addr[1] = memval & 0xff;
+#endif
+}
+#endif
+
+/****************************************************************************
+ * Name: imxrt_write32
+ *
+ * Description:
+ *   Write 32-bit little endian data
+ *
+ ****************************************************************************/
+
+#if 0							/* Not used */
+static void imxrt_write32(uint32_t memval, uint8_t *addr)
+{
+#ifdef CONFIG_ENDIAN_BIG
+	imxrt_write16(memval >> 16, &addr[0]);
+	imxrt_write16(memval & 0xffff, &addr[2]);
+#else
+	imxrt_write16(memval & 0xffff, &addr[0]);
+	imxrt_write16(memval >> 16, &addr[2]);
+#endif
+}
+#endif
+
+/****************************************************************************
+ * Name: imxrt_swap16
+ *
+ * Description:
+ *   Swap bytes on a 16-bit value
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ENDIAN_BIG
+static uint16_t imxrt_swap16(uint16_t value)
+{
+	return ((value >> 8) & 0xff) | ((value & 0xff) << 8);
+}
+#endif
+
+/****************************************************************************
+ * Name: imxrt_swap32
+ *
+ * Description:
+ *   Swap bytes on a 32-bit value
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_ENDIAN_BIG
+static uint32_t imxrt_swap32(uint32_t value)
+{
+	return (uint32_t) imxrt_swap16((uint16_t)((value >> 16) & 0xffff)) | (uint32_t) imxrt_swap16((uint16_t)(value & 0xffff)) << 16;
+}
+#endif
+
+/****************************************************************************
+ * Name: imxrt_printreg
+ *
+ * Description:
+ *   Print the contents of a LPC31 EHCI register
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_IMXRT_EHCI_REGDEBUG
+static void imxrt_printreg(volatile uint32_t *regaddr, uint32_t regval, bool iswrite)
+{
+	uinfo("%08x%s%08x\n", (uintptr_t) regaddr, iswrite ? "<-" : "->", regval);
```

Comment:
same as above. and please check at all places

Reply:
fixed it